src/printf.cpp

```diff
@@ -238,12 +239,17 @@ void process_printf(char*& data, const printf_descriptor_map_t& descs,
         next_part = part_end;
         arg_idx++;
     }
-
-    printf("%s", printf_out.str().c_str());
+    auto output = printf_out.str().c_str();
+    if (printf_cb != nullptr) {
+        printf_cb(output, buffer_size, data >= data_end, nullptr);
+    } else {
+        printf("%s", output);
+    }
```

Comment:
```suggestion
    auto output = printf_out.str();
    if (printf_cb != nullptr) {
        auto len = output.size();
        printf_cb(output.c_str(), len, data+len >= data_end, nullptr);
    } else {
        printf("%s", output.c_str());
    }
```
`printf_out.str()` returns a string. That will be destroyed if no handle is kept on it. Thus the current implementation has a `const char *` (through `c_str()`) on a buffer that is no longer allocated.

`len` and `complete` in `printf_cb` also seems to be incorrect in the current implementation

Reply:
thanks for that Romaric!